mailgun/tests.py

```diff
@@ -918,6 +918,215 @@ def test_multi_mailing_list_recipients_with_always_mail_staff_true(self, mock_ml
         )
         ml.send_mail.assert_has_calls([send_mail_call, send_mail_call])
 
+    @patch('mailgun.route_handlers.SuperSender.objects.filter')
+    @patch('mailgun.route_handlers.CourseInstance.objects.get_primary_course_by_canvas_course_id')
+    @patch('mailgun.route_handlers.MailingList.objects.get_or_create_or_delete_mailing_list_by_address')
+    def test_sender_receives_copy_when_always_mail_staff_false(self, mock_ml_get, mock_ci_get, mock_ss_filter):
+        """
+        TLT-2960
+        Verifies that when the  course_settings is False, the Sender gets a copy
+        of the email, even if they are not in the section
+        """
+        members = [{'address': a} for a in ['student@example.edu', 'unittest@example.edu']]
+        cs = MagicMock(always_mail_staff=False)
+
+        ml = MagicMock(
+            canvas_course_id=123,
+            section_id=456,
+            teaching_staff_addresses={'teacher1@example.edu', 'teacher2@example.edu'},
+            members=members,
+            address='class-list@example.edu',
+            course_settings=cs
+        )
+        mock_ml_get.return_value = ml
+
+        # prep a CourseInstance mock
+        ci = MagicMock(course_instance_id=789,
+                       canvas_course_id=ml.canvas_course_id,
+                       short_title='Lorem For Beginners',
+                       course=MagicMock(school_id='colgsas'))
+        mock_ci_get.return_value = ci
+
+        # prep the SuperSender result
+        mock_ss_filter.return_value.values_list.return_value=[]
+
+        # prep the post body
+
+        # send mail to a canvas section
+        recipients = (['canvas-123-456@example.edu'])
+        post_body = {
+            'sender': 'Unit Test <teacher1@example.edu>',
+            'recipient': recipients,
+            'subject': 'blah',
+            'body-plain': 'blah blah',
+            'To': recipients
+        }
+        post_body.update(generate_signature_dict())
+
+        # prep the request
+        request = self.factory.post('/', post_body)
+        request.user = self.user
+
+        # run the view, verify success
+        response = handle_mailing_list_email_route(request)
+        self.assertEqual(response.status_code, 200)
+        send_mail_call = call(
+            u'Unit Test via Canvas',
+            u'teacher1@example.edu',
+            ['teacher1@example.edu', 'unittest@example.edu', 'student@example.edu'],#sender is included in email
+            u'[Lorem For Beginners] blah',
+            attachments=[],
+            html='',
+            inlines=[],
+            message_id=None,
+            original_cc_address=[],
+            original_to_address=[u'canvas-123-456@example.edu'],
+            text=u'blah blah'
+        )
+        ml.send_mail.assert_has_calls([send_mail_call])
```

Comment:
FWIW, these built-in mock `assert`ions always scare me a bit. I'm always afraid we'll type them wrong or something, and they'll silently fail. I prefer to use `self.assertEqual()` and explicitly compare a value like `[call_args()](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args)`, to be paranoid.